### PR TITLE
Sale Screen UI fixes

### PR DESCRIPTION
**Pull request by brandon-lewis**
### Issues Fixed
https://github.com/JumpMind/openpos-framework/issues/2631

### Screenshots
When neither the Tax Exempt customer nor the Employee Discount are active, the sale items takes up the left section of the screen.
Screen Shot 2022-11-10 at 2 00 49 PM

We will only show the info that is active and relevant to the current transaction.
Screen Shot 2022-11-10 at 2 02 31 PM

Screen Shot 2022-11-10 at 2 02 43 PM

Employee Discount and Tax Exemption info was moved to the left, and Linked Customer membership pips were styled to be inline and scrollable to save on vertical space. Also notice that we no longer reserve the whole 275px for an image when a picture is not present on the sale item.
Screen Shot 2022-11-10 at 1 27 43 PM

Images still show as expected when the image is valid and present.
Screen Shot 2022-11-10 at 1 33 39 PM

The aria label for the item name/description currently shows when an image is not present, but this is a duplication of what already shows and the larger font ends up creating unnecessary wrapping. When an image is not valid/available on mobile, we now instead do not show the aria label.
Screen Shot 2022-11-10 at 1 29 17 PM

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/item-card/item-card.component.html [hunk 1]
```diff
@@ -1,110 +1,111 @@
-<div class="container" [class]="item.itemType?.itemType | lowercase"
```
Comment (brandon-lewis) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/item-card/item-card.component.html:1]: All I changed was line 12 to properly check for whether the image is available when determining the app-image size. IntelliJ still shows 12 as the only line change, so I'm not really sure what happened with this file...

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/item-card/item-card.component.ts [hunk 2]
```diff
@@ -1,103 +1,105 @@
...
+
+  @Input() enableHover = true;
+
+  public hover = false;
+
+  @ViewChild('kebab', {static: true}) kebab: KebabLabelButtonComponent;
+
+  isMobile$: Observable<boolean>;
+
+  imageExists: boolean = false;
```
Comment (brandon-lewis) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/item-card/item-card.component.ts:43]: Adding this imageExists property is the only change in this file. Again, not sure why Git thinks this entire file changed as IntelliJ still only reports this two-line change.

**Review by jrmihalick — APPROVED** [review]

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/sale/sale.component.scss [hunk 3]
```diff
@@ -23,6 +23,59 @@
...
+                    .customer-info-card {
+                        margin: 0;
+
+                        app-icon {
+                            padding: 4px;
+                        }
+
+                        .customer-info-card-header {
+                            display: flex;
+                            margin: -16px;
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/screens-with-parts/sale/sale.component.scss:51]: what is a 3 way negative margin with 1/4 padding and 1/4 border doing? This seems like a whacky way to position items

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/mobile-item/mobile-item.component.html [hunk 4]
```diff
@@ -1,8 +1,9 @@
 <div responsive-class class="mobile-item-outer">
     <div class="item-content">
-        <app-image *ngIf="item.imageUrl"
+        <app-image *ngIf="item.imageUrl && imageExists"
                    responsive-class
                    class="item-card-image"
+                   (imageExistsEvent)="updateImage($event)" [ngClass]="{'has-image': imageExists}"
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/mobile-item/mobile-item.component.html:6]: This is different than the other component where you did `imageExists = $event`.
Could you pick one method and stick to it?

**Review by jchirico8 — COMMENTED** [review]
@brandon-lewis if you can take a look at these comments and address them in the cherry pick that'll be fine

**Pull request merged**